iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -639,44 +664,48 @@ D2C_MESSAGE_HANDLE client_create_and_send_d2c(TEST_MESSAGE_CREATION_MECHANISM te
     EXPECTED_SEND_DATA* sendData = EventData_Create();
     ASSERT_IS_NOT_NULL(sendData, "Could not create the EventData associated with the event to be sent");
 
-    if (test_message_creation == TEST_MESSAGE_CREATE_BYTE_ARRAY)
-    {
-        msgHandle = IoTHubMessage_CreateFromByteArray((const unsigned char*)sendData->expectedString, strlen(sendData->expectedString));
-    }
-    else if (test_message_creation == TEST_MESSAGE_CREATE_STRING)
-    {
-        msgHandle = IoTHubMessage_CreateFromString(sendData->expectedString);
-    }
-    else
-    {
-        msgHandle = NULL;
-        ASSERT_FAIL("Unknown test message creation mechanism specified");
-    }
-    ASSERT_IS_NOT_NULL(msgHandle, "Could not create the D2C message to be sent");
-
-    MAP_HANDLE mapHandle = IoTHubMessage_Properties(msgHandle);
-    for (size_t i = 0; i < MSG_PROP_COUNT; i++)
+    if (sendData != NULL)
     {
-        if (g_e2e_test_options.use_special_chars)
+        if (test_message_creation == TEST_MESSAGE_CREATE_BYTE_ARRAY)
         {
-            if (Map_AddOrUpdate(mapHandle, MSG_PROP_KEYS_SPECIAL[i], MSG_PROP_VALS_SPECIAL[i]) != MAP_OK)
-            {
-                LogError("ERROR: Map_AddOrUpdate failed for property %zu!", i);
-            }
+            msgHandle = IoTHubMessage_CreateFromByteArray((const unsigned char*)sendData->expectedString, strlen(sendData->expectedString));
+        }
+        else if (test_message_creation == TEST_MESSAGE_CREATE_STRING)
+        {
+            msgHandle = IoTHubMessage_CreateFromString(sendData->expectedString);
         }
         else
         {
-            if (Map_AddOrUpdate(mapHandle, MSG_PROP_KEYS[i], MSG_PROP_VALS[i]) != MAP_OK)
+            msgHandle = NULL;
+            ASSERT_FAIL("Unknown test message creation mechanism specified");
+        }
+        ASSERT_IS_NOT_NULL(msgHandle, "Could not create the D2C message to be sent");
+
+        MAP_HANDLE mapHandle = IoTHubMessage_Properties(msgHandle);
```

Comment:
Nit - there's API's that make this code a bit easier, specifically `IoTHubMessage_SetProperty` which saves direct manipulation of the list.  A lot of older test code uses this `IoTHubMessage_Properties` semantic but it's officially deprecated.

Reply:
Also in the `IoTHubMessage_SetProperty` below for like "AzIoTHub_FaultOperationType", these end up ASSERT on the whole test.  Is there a reeason we only LogError but don't similarly ASSERT out of test in this path?